src/Featurizers/UnitTests/ShortGrainDropperFeaturizer_UnitTests.cpp

```diff
@@ -0,0 +1,190 @@
+
+    pTransformer->flush(callback);
+
+    CHECK(output == inferencingOutput);
+}
+
+TEST_CASE("Invalid Transformer/Estimator") {
+    //parameter setting
+    std::uint8_t windowSize = 0;
+    std::vector<std::uint8_t> lags = NS::TestHelpers::make_vector<std::uint8_t>();
+    std::uint8_t maxHorizon = 1;
+    nonstd::optional<std::uint8_t> cv = static_cast<std::uint8_t>(1);
+
+    CHECK_THROWS_WITH(NS::Featurizers::ShortGrainDropperEstimator<std::numeric_limits<size_t>::max()>(NS::CreateTestAnnotationMapsPtr(1), 2, windowSize, lags, maxHorizon, cv), "colIndex");
+    CHECK_THROWS_WITH(NS::Featurizers::ShortGrainDropperEstimator<std::numeric_limits<size_t>::max()>(NS::CreateTestAnnotationMapsPtr(1), 0, windowSize, lags, maxHorizon, cv), "lags");
+}
+
+TEST_CASE("Standard Test 1") {
+    std::vector<std::vector<std::vector<std::string>>> trainingBatches = NS::TestHelpers::make_vector<std::vector<std::vector<std::string>>>(
+        NS::TestHelpers::make_vector<std::vector<std::string>>(
+            NS::TestHelpers::make_vector<std::string>("a", "b"),
+            NS::TestHelpers::make_vector<std::string>("a", "b"),
+            NS::TestHelpers::make_vector<std::string>("a", "b"),
+            NS::TestHelpers::make_vector<std::string>("a", "b"),
+            NS::TestHelpers::make_vector<std::string>("a", "b"),
+            NS::TestHelpers::make_vector<std::string>("a", "c"),
+            NS::TestHelpers::make_vector<std::string>("a", "c"),
+            NS::TestHelpers::make_vector<std::string>("a", "c"),
+            NS::TestHelpers::make_vector<std::string>("a", "c"),
+            NS::TestHelpers::make_vector<std::string>("a", "d"),
+            NS::TestHelpers::make_vector<std::string>("a", "d"),
+            NS::TestHelpers::make_vector<std::string>("a", "d"),
+            NS::TestHelpers::make_vector<std::string>("c", "d")
+        )
+    );
+
+    std::vector<std::vector<std::string>> inferencingInput = NS::TestHelpers::make_vector<std::vector<std::string>>(
+        NS::TestHelpers::make_vector<std::string>("a", "b"),
+        NS::TestHelpers::make_vector<std::string>("a", "c"),
+        NS::TestHelpers::make_vector<std::string>("a", "d"),
+        NS::TestHelpers::make_vector<std::string>("c", "d"),
+        NS::TestHelpers::make_vector<std::string>("e", "f")
+    );
+
+    std::vector<bool> inferencingOutput = {
+        false,
+        true,
+        true,
+        true,
+        false
+    };
+
+    //parameter setting
+    std::uint8_t windowSize = 0;
+    std::vector<std::uint8_t> lags = NS::TestHelpers::make_vector<std::uint8_t>(static_cast<std::uint8_t>(0), static_cast<std::uint8_t>(0));
+    std::uint8_t maxHorizon = 1;
+    nonstd::optional<std::uint8_t> cv = static_cast<std::uint8_t>(1);
+
+    TestImpl(
+        trainingBatches,
+        inferencingInput,
+        inferencingOutput,
+        windowSize,
+        lags,
+        maxHorizon,
+        cv
+    );
+}
+
+TEST_CASE("Standard Test 2") {
```

Comment:
>2 [](start = 25, length = 1)

How is this test different from "Standard Test"? Consider a more descriptive name for the test.